### PR TITLE
New ETL Geometry Validation

### PR DESCRIPTION
**Pull request by gsorrentino18**
#### PR description:
This PR modifies the ETL plugins in order to validate the new ETL-2 disks geometry, following PR #31765 prepared by @fabiocos. It still provides monitoring histograms for the old ETL-1 disk geometry, distinguishing between the two geometries via MTDTopologyMode.

#### PR validation:

The new code has been tested in CMSSW_11_2_0_pre7. It needs the full 2-Disks geometry implemented in #31765 to be tested on 2-Disks scenarios, and #31654 to produce meaningful results for the new geometry. 
The size of the new histograms is:
962.49 KiB       MTD/ETL
238.40 KiB       MTD/BTL
18.05 KiB        MTD/GlobalReco
with respect to previous size:
481.31 KiB       MTD/ETL
238.40 KiB       MTD/BTL
12.77 KiB        MTD/GlobalReco

### CODE REVIEW

**Comment by jfernan2** [comment]
please test with #31765,#31654

File: Validation/MtdValidation/plugins/MtdGlobalRecoValidation.cc [hunk 1]
```diff
@@ -98,6 +94,17 @@ void MtdGlobalRecoValidation::analyze(const edm::Event& iEvent, const edm::Event
   using namespace geant_units::operators;
   using namespace std;
 
+  edm::ESHandle<MTDTopology> topologyHandle;
+  iSetup.get<MTDTopologyRcd>().get(topologyHandle);
+  const MTDTopology* topology = topologyHandle.product();
+
+  bool topo1Dis = false;
+  bool topo2Dis = false;
+  if (topology->getMTDTopologyMode() <= static_cast<int>(MTDTopologyMode::Mode::barphiflat))
```
Comment (fabiocos) [inline Validation/MtdValidation/plugins/MtdGlobalRecoValidation.cc:103]: here as well

File: Validation/MtdValidation/plugins/EtlSimHitsValidation.cc [hunk 2]
```diff
@@ -104,21 +108,56 @@ void EtlSimHitsValidation::analyze(const edm::Event& iEvent, const edm::EventSet
   iSetup.get<MTDDigiGeometryRecord>().get(geometryHandle);
   const MTDGeometry* geom = geometryHandle.product();
 
+  edm::ESHandle<MTDTopology> topologyHandle;
+  iSetup.get<MTDTopologyRcd>().get(topologyHandle);
+  const MTDTopology* topology = topologyHandle.product();
+
+  bool topo1Dis = false;
+  bool topo2Dis = false;
+  if (topology->getMTDTopologyMode() <= static_cast<int>(MTDTopologyMode::Mode::barphiflat))
```
Comment (fabiocos) [inline Validation/MtdValidation/plugins/EtlSimHitsValidation.cc:117]: here as well

File: Validation/MtdValidation/plugins/EtlLocalRecoValidation.cc [hunk 3]
```diff
@@ -92,6 +95,17 @@ EtlLocalRecoValidation::~EtlLocalRecoValidation() {}
...
 void EtlLocalRecoValidation::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
   using namespace edm;
+  using namespace std;
+  edm::ESHandle<MTDTopology> topologyHandle;
+  iSetup.get<MTDTopologyRcd>().get(topologyHandle);
+  const MTDTopology* topology = topologyHandle.product();
+
+  bool topo1Dis = false;
+  bool topo2Dis = false;
+  if (topology->getMTDTopologyMode() <= static_cast<int>(MTDTopologyMode::Mode::barphiflat))
```
Comment (fabiocos) [inline Validation/MtdValidation/plugins/EtlLocalRecoValidation.cc:105]: here as well

**Comment by jfernan2** [comment]
@gsorrentino18  is there any reason to have all the *D2 histograms empty?

**Comment by fabiocos** [comment]
@jfernan2 just discussed with @gsorrentino18 this should not happen in principle. As the former test of #31654 is ok that should not be the culprit (the code update is quite transparent). I do not see at first sight as well how my PR could cause this, but the best way to test is to launch again the tests for each PR separately, solve the issue where it is, and then go back to the combined test.

**Comment by fabiocos** [comment]
please test

**Comment by fabiocos** [comment]
@gsorrentino18 the test confirms that the issue is with your code, please investigate

File: Validation/MtdValidation/plugins/EtlLocalRecoValidation.cc [hunk 4]
```diff
@@ -267,7 +574,8 @@ void EtlLocalRecoValidation::fillDescriptions(edm::ConfigurationDescriptions& de
   desc.add<std::string>("folder", "MTD/ETL/LocalReco");
   desc.add<edm::InputTag>("recHitsTag", edm::InputTag("mtdRecHits", "FTLEndcap"));
   desc.add<edm::InputTag>("recCluTag", edm::InputTag("mtdClusters", "FTLEndcap"));
-  desc.add<double>("hitMinimumEnergy", 1.);  // [MeV]
+  desc.add<double>("hitMinimumEnergy1Dis", 0.01);   // [MeV]
```
Comment (fabiocos) [inline Validation/MtdValidation/plugins/EtlLocalRecoValidation.cc:577]: @gsorrentino18 this threshold has changed compared to the default version also fo the old scenarios, why?

File: Validation/MtdValidation/plugins/EtlSimHitsValidation.cc [hunk 5]
```diff
@@ -264,7 +540,8 @@ void EtlSimHitsValidation::fillDescriptions(edm::ConfigurationDescriptions& desc
 
   desc.add<std::string>("folder", "MTD/ETL/SimHits");
   desc.add<edm::InputTag>("inputTag", edm::InputTag("mix", "g4SimHitsFastTimerHitsEndcap"));
-  desc.add<double>("hitMinimumEnergy", 0.1);  // [MeV]
+  desc.add<double>("hitMinimumEnergy1Dis", 0.01);   // [MeV]
```
Comment (fabiocos) [inline Validation/MtdValidation/plugins/EtlSimHitsValidation.cc:543]: again why this change?

Comment (gsorrentino18) [reply Validation/MtdValidation/plugins/EtlSimHitsValidation.cc:543]: My mistake, for the old scenario the thresholds should not change.

**Comment by fabiocos** [comment]
please test

**Comment by gsorrentino18** [comment]
> Pull request #31769 was updated. @andrius-k, @kmaeshima, @ErnestaP, @cmsbuild, @jfernan2, @fioriNTU can you please check and sign again.

@jfernan2 @fabiocos the issues in the comparisons should be fixed now. Could you please test again?

**Comment by jfernan2** [comment]
please test

**Comment by fabiocos** [comment]
please test

the error looks unrelated to this PR, apparently a glitch in recovering inputs, restart

**Comment by fabiocos** [comment]
@smuzaffar a problem with the comparison process?

**Comment by smuzaffar** [comment]
@fabiocos , no just a large back log of job in jenkins. We have issues with CVMFS publisher last night which was only fixed this morning at 10. This caused a large backlog of job

**Comment by fabiocos** [comment]
@smuzaffar ok, thanks for the explanation. 11 hours for comparison is definitely out of normal

**Comment by fabiocos** [comment]
The residual differences have nothing to do with this PR, 2017 workflows cannot be affected by MTD validation code

**Comment by jfernan2** [comment]
> @gsorrentino18 is there any reason to have all the *D2 histograms empty?

This issue seems to be there still, is it understood? 
There are also MEs in  MTD/GlobalReco with zero efficiency, some are *D1 and some are *D2

**Comment by gsorrentino18** [comment]
> > @gsorrentino18 is there any reason to have all the *D2 histograms empty?
> 
> This issue seems to be there still, is it understood?
> There are also MEs in MTD/GlobalReco with zero efficiency, some are *D1 and some are *D2

@jfernan2 thanks, you're right. I'm investigating the issue

**Comment by gsorrentino18** [comment]
> Pull request #31769 was updated. @andrius-k, @kmaeshima, @ErnestaP, @cmsbuild, @jfernan2, @fioriNTU can you please check and sign again.

@jfernan2 There were indeed some issues in filling some *D1 histograms. However, the *D2 histograms are always empty testing the PR with D49 workflow, since in this scenario the ETL has only one disc. The *D2 distributions are filled only when testing using 2-discs ETL scenarios.

**Comment by fabiocos** [comment]
@gsorrentino18 did you verify that everything in D49 now looks as before, apart for the change in histogram name?

**Comment by fabiocos** [comment]
hold

pending confirmation by @gsorrentino18

**Comment by fabiocos** [comment]
@gsorrentino18 could you please summarize the updates and the result of checks?

**Comment by gsorrentino18** [comment]
> @gsorrentino18 could you please summarize the updates and the result of checks?

I checked the *D1 histograms to be sure that they are all correctly filled, and that for D49 the distributions look as before, except for a change in name. I also adjust the range of the hit Z position histograms in order to have the same range of the previous distributions and to exactly compare them in D49 scenario. The PR is now ready for test.

**Comment by fabiocos** [comment]
unhold

**Comment by fabiocos** [comment]
please test

**Comment by jfernan2** [comment]
+1

**Comment by silviodonato** [comment]
+1
